src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,404 @@
+        that the optimizer was created after the model was wrapped with :meth:`setup_module` with a reference to the
+        flattened parameters.
+        """
+        if _TORCH_GREATER_EQUAL_2_0:
+            return optimizer
+
+        from torch_xla.distributed.fsdp.xla_flatten_params_wrapper import FlatParameter
+
+        num_groups = len(optimizer.param_groups)
+        if num_groups > 1:
+            raise ValueError(
+                "An optimizer used with an XLA FSDP model does not support multiple param groups."
+                f" Found {num_groups} parameter groups."
+            )
+
+        if any(isinstance(param, FlatParameter) for param in optimizer.param_groups[0]["params"]):
+            return optimizer
+
+        raise ValueError(
+            "The optimizer does not seem to reference any XLA FSDP parameters. HINT: Make sure to create the optimizer"
+            " after setting up the model."
+        )
+
+    def optimizer_step(
+        self,
+        optimizer: Optimizable,
+        **kwargs: Any,
+    ) -> Any:
+        """Overrides default tpu optimizer_step since FSDP should not call
+        `torch_xla.core.xla_model.optimizer_step`. Performs the actual optimizer step.
+
+        Args:
+            optimizer: the optimizer performing the step
+            **kwargs: Any extra arguments to ``optimizer.step``
+        """
+        loss = optimizer.step(**kwargs)
+        import torch_xla.core.xla_model as xm
+
+        xm.mark_step()
+        return loss
+
+    def module_to_device(self, module: Module) -> None:
+        module.to(self.root_device)
+
+    def clip_gradients_norm(
+        self,
+        module: Module,
+        optimizer: Optimizer,
+        max_norm: Union[float, int],
+        norm_type: Union[float, int] = 2.0,
+        error_if_nonfinite: bool = True,
+    ) -> Tensor:
+        """Clip gradients by norm."""
+        from torch_xla.distributed.fsdp import XlaFullyShardedDataParallel as XLAFSDP
+
+        if not isinstance(module, XLAFSDP):
+            # the root must be wrapped
+            raise TypeError(
+                "Gradient clipping with XLAFSDP is only possible if the module passed to"
+                f" `{self.__class__.__name__}.clip_gradients_norm` is wrapped in `XLAFullyShardedDataParallel`."
+                f" Got: {module.__class__.__name__}."
+            )
+        self.precision.unscale_gradients(optimizer)
+        return module.clip_grad_norm_(max_norm=max_norm, norm_type=norm_type)
+
+    def clip_gradients_value(  # type: ignore[override]
+        self, module: Module, optimizer: Optimizer, clip_val: Union[float, int]
+    ) -> None:
+        """Clip gradients by value."""
+        raise NotImplementedError(
+            "XLA's FSDP strategy does not support to clip gradients by value."
+            " Consider clipping by norm instead or choose another strategy!"
+        )
+
+    def save_checkpoint(
```

Comment:
Should this method `mark_step` before saving?

Reply:
Added